substrafl_examples/get_started/plot_substrafl_sklearn_fedavg.py

```diff
@@ -0,0 +1,523 @@
+
+    client = clients[org_id]
+
+    # Add the training data on each organization.
+    data_sample = DataSampleSpec(
+        data_manager_keys=[dataset_keys[org_id]],
+        test_only=False,
+        path=data_path / f"org_{ind+1}" / "train",
+    )
+    train_datasample_keys[org_id] = client.add_data_sample(
+        data_sample,
+        local=True,
+    )
+
+    # Add the testing data on each organization.
+    data_sample = DataSampleSpec(
+        data_manager_keys=[dataset_keys[org_id]],
+        test_only=True,
+        path=data_path / f"org_{ind+1}" / "test",
+    )
+    test_datasample_keys[org_id] = client.add_data_sample(
+        data_sample,
+        local=True,
+    )
+
+# %%
+# Metrics registration
+# ====================
+#
+# A metric corresponds to an algorithm used to compute the score of predictions on a
+# **datasample**.
+# Concretely, a metric corresponds to an archive *(tar or zip file)*, automatically build
+# from:
+#
+# - a **Python scripts** that implement the metric computation
+# - a `Dockerfile <https://docs.docker.com/engine/reference/builder/>`__ to specify the required dependencies of the
+#   **Python scripts**
+
+import zipfile
+
+from substra.sdk.schemas import AlgoInputSpec
+from substra.sdk.schemas import AlgoOutputSpec
+from substra.sdk.schemas import AlgoSpec
+from substra.sdk.schemas import AssetKind
+
+inputs_metrics = [
+    AlgoInputSpec(
+        identifier="datasamples",
+        kind=AssetKind.data_sample,
+        optional=False,
+        multiple=True,
+    ),
+    AlgoInputSpec(identifier="opener", kind=AssetKind.data_manager, optional=False, multiple=False),
+    AlgoInputSpec(identifier="predictions", kind=AssetKind.model, optional=False, multiple=False),
+]
+
+outputs_metrics = [AlgoOutputSpec(identifier="performance", kind=AssetKind.performance, multiple=False)]
+
+objective = AlgoSpec(
+    inputs=inputs_metrics,
+    outputs=outputs_metrics,
+    name="Accuracy",
+    description=assets_directory / "metric" / "description.md",
+    file=assets_directory / "metric" / "metrics.zip",
+    permissions=permissions,
+)
+
+METRICS_DOCKERFILE_FILES = [
+    assets_directory / "metric" / "iris_metrics.py",
+    assets_directory / "metric" / "Dockerfile",
+]
+
+archive_path = objective.file
+with zipfile.ZipFile(archive_path, "w") as z:
+    for filepath in METRICS_DOCKERFILE_FILES:
+        z.write(filepath, arcname=filepath.name)
+
+metric_key = clients[ALGO_ORG_ID].add_algo(objective)
+
+# %%
+# Specify the machine learning components
+# ***************************************
+# In this section, you will register an algorithm and its dependencies,
+# and specify the federated learning strategy
+# as well as the nodes on which to train and to test.
+
+# %%
+# Model definition
+# ================
+# We choose to use a logistic regression as the model to train.
+
+from sklearn import linear_model
+
+cls = linear_model.LogisticRegression(random_state=SEED, warm_start=True, max_iter=3)
```

Comment:
Add a comment on warm_start as it's not a well known parameter?
And why max_iter=3?

Reply:
I added comment on these two